src/utils/compat.ts

```diff
@@ -83,6 +86,33 @@ const decodeAddress = (asset: string, addr: string): DecodedAddress => {
     };
 };
 
+const probeAddress = (address: string): string | null => {
+    let asset: string | null = null;
+    btcPrefixes.forEach((prefix: string) => {
+        if (address.startsWith(prefix)) {
+            asset = BTC;
+        }
+    });
+    liquidPrefixes.forEach((prefix: string) => {
+        if (address.startsWith(prefix)) {
+            asset = LBTC;
+        }
+    });
+    if (asset !== null) {
+        return asset;
+    }
+    // fallback to more costly decoding
+    try {
+        decodeAddress(BTC, address);
+        return BTC;
+    } catch (e) {}
+    try {
+        decodeAddress(LBTC, address);
+        return LBTC;
+    } catch (e) {}
```

Comment:
We need to decode the address anyways. Why don't we start decoding with the set network and then try all the others known to us?

- if address is valid for set network -> address valid
- address is valid for other network -> address valid & switch network
- address is not valid for any network -> address invalid

Reply:
we  dont need to decode the address when pasting into the `InvoiceInput` and `setOnchainAddress` would decode it twice